src/python/turicreate/toolkits/object_detector/_tf_image_augmenter.py

```diff
@@ -22,20 +22,27 @@
 
 tf.disable_v2_behavior()
 
-def get_augmented_data(images, annotations, output_height, output_width, resize_only):
+def get_resized_data(images, output_height, output_width):
 
     # Suppresses verbosity to only errors
     tf.logging.set_verbosity(tf.logging.ERROR)
 
-    session = tf.Session()
     output_shape = (output_height, output_width)
-    
-    if resize_only:
+
+    with tf.Session() as session:
         images = get_resized_images(images, output_shape)
         resized_images = session.run(images)
         resized_images = np.array(resized_images, dtype=np.float32)
         return tuple((resized_images, len(resized_images)*[np.zeros(6)]))
-    else:
+
+def get_augmented_data(images, annotations, output_height, output_width):
+
+    output_shape = (output_height, output_width)
+
+    # Suppresses verbosity to only errors
+    tf.logging.set_verbosity(tf.logging.ERROR)
+
+    with tf.Session() as session:
```

Comment:
Could you have any explanation of this? It seems very rare that TF won't recycle unused sessions.

Reply:
On digging a bit, I have found resources that state that session leaks may occur if sessions are not closed properly. I am trying to understand a bit more by reproducing it and looking at the memory footprint.